api/app/fireweather_bot/hourly_actuals.py

```diff
@@ -21,102 +25,124 @@
-        data_df.drop(index=data_df[~data_df['display_name'].isin(
-            station_codes.keys())].index, inplace=True)
-        # replace 'display_name' column (station name) in df with station_id
-        # and rename the column appropriately
-        data_df['display_name'].replace(station_codes, inplace=True)
-        data_df.rename(columns={'display_name': 'station_code'}, inplace=True)
-
-        # write to database using _session's engine
+        return self.now
+
+    async def run_wfwx(self):
+        """ Entry point for running the bot """
+        async with ClientSession(connector=TCPConnector(limit=100)) as session:
+            header = await wildfire_one.get_auth_header(session)
+
+            start_date = self._get_start_date()
+            end_date = self._get_end_date()
+
+            station_hourly_readings = await wildfire_one.get_hourly_readings_all_stations(
+                session, header, start_date, end_date)
+
+        logger.info("Station hourly readings: %s", station_hourly_readings)
+
         with app.db.database.get_write_session_scope() as session:
-            # write the data_df to the database one row at a time, so that if data_df contains >=1 rows that
-            # are duplicates of what is already in the db, the write won't fail for the unique rows
-            # NOTE: iterating over the data_df one Series (row) at a time is ugly, but until pandas is
-            # updated with a fix, this is the easiest work-around.
-            # See https://github.com/pandas-dev/pandas/issues/15988
-            # pylint: disable=unused-variable
-            for index, row in data_df.iterrows():
-                try:
-                    # Go from pandas to a python dict.
-                    data = row.to_dict()
-                    # Format and fix timestamp, make it be tz aware.
-                    data['weather_date'] = _fix_datetime(data['weather_date'])
-                    # Throw the data into the model, and persist in the database
-                    save_hourly_actual(session, HourlyActual(**data))
-                except IntegrityError:
-                    logger.info('Skipping duplicate record for %s @ %s',
-                                data['station_code'], data['weather_date'])
-                    session.rollback()
-
-    def construct_request_body(self):
-        return {
-            'Start_Date': self._get_start_date(),
-            'End_Date': self._get_end_date(),
-            'Format': 'CSV',
-            'cboFilters': config.get('BC_FIRE_WEATHER_FILTER_ID'),
-            'rdoReport': 'OSBH',
-        }
+            for station_hourly_reading in station_hourly_readings:
+                for hourly_reading in station_hourly_reading.values:
+                    hourly_actual = parse_hourly_actual(
+                        station_hourly_reading.station.code, hourly_reading)
+                    if hourly_actual is not None:
+                        save_hourly_actual(session, hourly_actual)
```

Comment:
You need to handle duplicates, right now it's failing on duplicates, killing the cronjob and logging to rocket.